DeviceThread に Logger を追加

brly opened the pull request:
DeviceThread でログ出力を行いたいため `Device` `DeviceBuilder` `DeviceThread` の spawn 時に Logger を必ず取るように変更しました。

もともと DeviceThread の `Command::Put` 処理時の lump_id を知りたかったというものがあり、この PR でもその追加に留まっています。

また、ログ出力時にどの DeviceThread からの出力なのかを識別するために、かなり大雑把な `id` という文字列を追加して、合わせて出力するようにしています。frugalos からは一つのプロセスで複数の DeviceThread を利用していますが、識別する際は DeviceId を利用していて、その情報がログに欲しいと思ったためです。

brly committed:
Add Logger

koba-e964 commented on src/device/thread.rs:
```diff
@@ -40,6 +43,8 @@ where
         builder: DeviceBuilder,
         init_storage: F,
+        logger: Logger,
+        id: Option<String>,
```
`id` が基本的にデバイスの ID を指すのであれば、`device_id` みたいな名前にしてもよいのでは、と思います

brly replied:
frugalos から参照する時はそうなのですが、仮に別のライブラリから参照する時は DeviceThread としての id として渡したい時もあるかも？と少し思ったりしたので、抽象的な id にしていました。

Device の Id でも DeviceThread の Id でも大きく変わらない気はしていますが..。

brly replied:
提案を受けて id を足すのは取りやめます。ログ追加のために足すのは前向きではなかったので。

shinnya commented on src/device/thread.rs:
```diff
@@ -40,6 +43,8 @@ where
     pub fn spawn<F>(
         builder: DeviceBuilder,
         init_storage: F,
+        logger: Logger,
```
spawn のインターフェースを変えるのではなく、 `DeviceBuilder` に設定できるようにしてはどうでしょうか。参考 https://github.com/sile/fibers_rpc/blob/6b631ee5957fad8910df625df49d39e12e4e09ff/src/client_service.rs#L42

また、`id` は logger に設定する仕様にした方が引き回さなくてよくなるため使い勝手がよさそうです。参考 https://github.com/slog-rs/slog/blob/f4ada8ef4992f0ba3ff139d883319453bac8d70d/examples/named.rs#L11

brly replied:
id はもともとログ出力のために足すのはやりたくなく、logger の機能で出力に足せるのであれば、それを使うようにしてみます。

DeviceBuilder に設定するやり方で、もとの spawn IF を変えずに済みそうなのでそうしてみます。

brly committed:
Remove DeviceThread.id and Fix DeviceBuilder

Apply a amendment (https://github.com/frugalos/cannyls/pull/38#discussion_r315998454)

shinnya commented on src/device/thread.rs:
```diff
@@ -70,6 +71,7 @@ where
                     start_busy_time: None,
                     command_tx,
                     command_rx,
+                    logger,
```
(変更途中かもしれませんが) `DeviceBuilder` に `logger` を設定できるようにしたのであれば、`DeviceBuilder` が保持している `logger` を参照するのが自然に思いました。

brly replied:
見落としておりました、直します

brly committed:
Fix DeviceThread::spawn interfaces

brly committed:
Remove unused macro
